lib/pushbox.js

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const Pool = require('./pool')
+const {URLSearchParams} = require('url')
+
+const LOG_OP_RETRIEVE = 'pushbox.retrieve'
+const LOG_OP_STORE = 'pushbox.store'
+
+const SERVICES_TTL = {
+  sendtab: 30 * 24 * 3600 // 30 days.
+}
+
+module.exports = function (log, config) {
+  const pool = new Pool(config.pushboxUrl, { timeout: 1000 })
+
+  return {
+    /**
+     * Retrieves messages for a specific device. Relays the request to Pushbox.
+     *
+     * @param {String} uid - Firefox Account uid
+     * @param {String} deviceId
+     * @param {string} service - Pushbox service identifier
+     * @param {Object} options
+     * @param {String} [options.index]
+     * @param {Number} [options.limit]
+     * @param {String} oAuthToken - Hex string of the user's Pushbox bearer token
+     * @returns {Promise}
+     */
+    retrieve (uid, deviceId, service, {index, limit}, oAuthToken) {
+      log.trace({
+        op: LOG_OP_RETRIEVE,
+        uid,
+        deviceId,
+        service,
+        index,
+        limit
+      })
+      const params = new URLSearchParams()
+      if (index !== undefined) {
```

Comment:
can we just do `if (index)` ?

Reply:
Index might be 0, can't!

Reply:
right!

Reply:
According to the docstring and the validators on `/account/device/messages`, `index` is a string and so should never be falsy, even if it's `"0"`.